Dockerfile

```diff
@@ -14,12 +14,38 @@ COPY bin /bin
 FROM ${DEVICEOS_BASE_IMAGE}:${DEVICEOS_VERSION} as deviceos
 WORKDIR /
 
-# Main image combining base image and DeviceOS sources
+# Base image
+FROM ${BUILDPACK_BASE}:${BUILDPACK_BASE_VERSION} as builder
+
+RUN \
+  apt-get update -q \
+  && apt-get install -qy wget vim-common parallel libarchive-zip-perl \
+  && curl https://prtcl.s3.amazonaws.com/install-apt.sh | sh \
+  && apt-get clean \
+  && apt-get purge \
+  && rm -rf /var/lib/apt/lists/* /tmp/* /var/tmp/* \
+  && cat /etc/*release
+
+COPY --from=deviceos / /firmware/
+ENV PATH="/root/.particle/bin":$PATH
+RUN \
+  prtcl toolchain:install source:/firmware \
+  && prtcl toolchain:use source:/firmware \
+  && echo ":::: Using $(which arm-none-eabi-gcc)" \
+  && echo ":::: With directories and files" \
+  && ls -la /root/.particle \
+  && ls -la /root/.particle/toolchains \
+  && ls -la /root/.particle/bin
+
+# Main image combining base image, toolchain, and DeviceOS sources
 # Some sensible default
 FROM ${DOCKER_IMAGE_NAME}:${BASE_VERSION} as main
 ENV FIRMWARE_REPO=not-used
 WORKDIR /
 COPY --from=deviceos / /firmware/
+COPY --from=builder /root/.particle /root/.particle
+ENV PATH="/root/.particle/bin":$PATH
+
```

Comment:
Can we perhaps combine these two steps, since there is anyway a dependency on Device OS sources and there is no benefit in caching etc?

Reply:
works for me 👍 

fwiw we were trying to execute the proposal over in [ch76129](https://app.clubhouse.io/particle/story/76129/update-firmware-buildpack-builder-to-use-workbench-bundles) but ultimately i defer to you all.